code/src/sixsq/nuvla/server/resources/infrastructure_service.clj

```diff
@@ -169,14 +187,32 @@ existing infrastructure-service-template resource.
   (edit-impl request))
 
 
-(def delete-impl (std-crud/delete-fn resource-type))
+(defn delete-impl
+  [{{uuid :uuid} :params :as request}]
+  (try
+    (-> (str resource-type "/" uuid)
+        (db/retrieve request)
+        infra-service-utils/verify-can-delete
+        (a/can-modify? request)
+        (db/delete request))
+    (catch Exception e
+      (let [response (ex-data e)
+            status (:status response)]
+        (if (= 412 status)
```

Comment:
This seems to be the wrong behavior to me.  You've checked that the infrastructure service is in a state where it cannot be deleted, yet you then create a job to do exactly that. Instead, an action should be used to put the job into a "deletable" state and only then should a job for the clean up be created.

Reply:
See general comment about state management.